src/components/diff/Diff.tsx

```diff
@@ -0,0 +1,54 @@
+import * as React from 'react';
+import { IDiffContext } from './model';
+import { IRenderMimeRegistry } from '@jupyterlab/rendermime';
+import { NBDiff } from './NbDiff';
+import { PathExt } from '@jupyterlab/coreutils';
+
+/**
+ * A registry which maintains mappings of file extension to diff provider components.
+ */
+const DIFF_PROVIDER_REGISTRY = {
+  '.ipynb': NBDiff
+};
+
+/**
+ * Determines if a given file is supported for diffs.
+ *
+ * This will be removed when "Plaintext" diffs are supported since that will be used
+ * for cases where there is not a dedicated diff provider.
+ *
+ * @param path the file path
+ */
+export function isDiffSupported(path: string): boolean {
+  return PathExt.extname(path).toLocaleLowerCase() in DIFF_PROVIDER_REGISTRY;
+}
+
+export interface IDiffProps {
+  renderMime: IRenderMimeRegistry;
+  path: string;
+  diffContext: IDiffContext;
+}
+
+/**
+ * The parent diff component which maintains a registry of various diff providers.
+ * Based on the extension of the file, it delegates to the relevant diff provider.
+ */
+export class Diff extends React.Component<IDiffProps, {}> {
```

Comment:
This class is a bit strange since it doesn't do much else except rendering another component. I can't help but think it's unnecessary, but an alternative solution isn't immediately obvious. It could be stateless: https://itnext.io/react-component-class-vs-stateless-component-e3797c7d23ab. Does this help in this case? https://reactjs.org/docs/composition-vs-inheritance.html

Reply:
Yeah this should just be a functional stateless component. 

```typescript
function Diff(props: IDiffProps) {
   ...
}
```

Reply:
In terms of necessity, I do think we need this top-level component to delegate to a filetype-specific component and maintain the registry of various diff providers. For e.g., the next thing on the plate is to add `<PlainTextDiff />` and it'll invoke that for non-ipynb files.

I'm good to make it a stateless component for less verbosity.